src/wechaty/user/contact.py

```diff
@@ -163,27 +168,31 @@ def is_ready(self):
         """
         return self.payload is not None and self.payload.name is not None
 
-    def ready(self):
+    async def ready(self, force_sync: bool = False):
         """
         load contact object from puppet
         :return:
         """
         log.info('load contact %s', self.name)
-        try:
-            self.payload = self.puppet.get_contact_payload(self.contact_id)
-        except Exception as e:
-            log.info("can't load contact %s payload, message : %s",
-                     self.name,
-                     str(e.args))
+        if force_sync or self.is_ready():
```

Comment:
I believe what you want to do is:

```diff
-  if force_sync or self.is_ready():
+  if force_sync or !self.is_ready():
```

And I'd like to change to the following flow logic:

```py
if self.is_ready() and not force_sync:
  return

// Do the sync staffs
```

Because the above flow logic can save us for indent for the following codes.

Reply:
I find this bug just now. I have change it later.  
Thank you for giving me so detailed code view.

Reply:
I find this bug just now. I have changed it.
Thank you for giving me so detailed code view.